File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.go
func digitProduct(n int) int{
    if n < 10{
        return n;
    }
    x := n/10
    y := n%10
    return (n*n - y*y - 100*x*x)/20
}
func smallestNumber(n int, t int) int {
    for(digitProduct(n) % t != 0){
        n++
    }
    return n
}

File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.cpp
class Solution {
public:
    int digitProduct(int n){
        if(n<10) return n;
        int x = n/10;
        int y = n%10;
        int xy = (n*n - y*y - 100*x*x)/20;
        return xy;
    }
    int smallestNumber(int n, int t) {
        //small constraints => brute force
        while(digitProduct(n)%t != 0){
            n++;
        }
        return n;
    }
};